internal/handlers/common/filter.go

```diff
@@ -511,3 +520,186 @@ func filterFieldExprExists(fieldExist bool, exprValue any) (bool, error) {
 		return false, nil
 	}
 }
+
+// filterFieldExprType handles {field: {$type: value}} filter.
+func filterFieldExprType(fieldValue, exprValue any) (bool, error) {
+	switch exprValue := exprValue.(type) {
+	case string:
+		res, err := matchTypeByAlias(fieldValue, exprValue)
+		if err != nil {
+			return res, err
+		}
+		return res, nil
+
+	case int32:
+		alias, err := getTypeAliasByCode(exprValue)
+		if err != nil {
+			return false, err
+		}
+		res, err := matchTypeByAlias(fieldValue, alias)
+		if err != nil {
+			return res, err
+		}
+		return res, nil
+
+	case *types.Array:
+		hasSameType := exprValue.HasSameTypeElements()
+
+		for i := 0; i < exprValue.Len(); i++ {
+			aliasValue, err := exprValue.Get(i)
+			if err != nil {
+				panic(err)
+			}
+
+			switch alias := aliasValue.(type) {
+			case string:
+				res, err := matchTypeByAlias(fieldValue, alias)
+				if err != nil {
+					return false, err
+				}
+				if res {
+					return true, nil
+				}
+
+			case int32:
+				aliasCode, err := getTypeAliasByCode(alias)
+				if err != nil {
+					return false, err
+				}
+
+				if !hasSameType {
+					continue
+				}
+
+				res, err := matchTypeByAlias(fieldValue, aliasCode)
+				if err != nil {
+					return false, err
+				}
+				if res {
+					return true, nil
+				}
+			case float64:
+				if math.IsNaN(alias) || math.IsInf(alias, 0) {
+					return false, NewErrorMsg(ErrBadValue, `Invalid numerical type code: `+
+						strings.Trim(strings.ToLower(fmt.Sprintf("%v", alias)), "+"))
+				}
+				if alias != math.Trunc(alias) {
+					return false, NewErrorMsg(ErrBadValue, fmt.Sprintf(`Invalid numerical type code: %v`, alias))
+				}
+
+				aliasCode, err := getTypeAliasByCode(int32(alias))
+				if err != nil {
+					return false, err
+				}
+
+				if !hasSameType {
+					continue
+				}
+
+				res, err := matchTypeByAlias(fieldValue, aliasCode)
+				if err != nil {
+					return false, err
+				}
+				if res {
+					return true, nil
+				}
+			default:
+				return false, NewErrorMsg(
+					ErrBadValue,
+					fmt.Sprintf(
+						`Invalid numerical type code: %s`, aliasValue),
+				)
+			}
+		}
+		return false, nil
+
+	default:
+		return false, NewErrorMsg(
+			ErrBadValue,
+			fmt.Sprintf(
+				`Invalid numerical type code: %v`, exprValue),
```

Comment:
The same formatting issue I mention every time

Reply:
Fixed